### ISSUE TITLE
Create data structure to capture errors

### ISSUE THREAD

**Issue by alanz** [JPMoresmau/error-analyze#1]
Silly issue title, but it would be great if we can look at the data structures you end up with and use them to understand what changes need to happen to GHC/Cabal to be able to emit them directly

**Comment by alanz** [comment]
See also https://github.com/commercialhaskell/stack-ide/blob/c7af3770cdbe5550094e65de12358db2a5b191ce/stack-ide/src/Stack/Ide/AnnotateMessage.hs

**Comment by mgsloan** [comment]
Agreed, I also wrote that error message regexer with that in mind - that eventually GHC will provide such info without it needing to be parsed out.
